psi4/src/psi4/libfock/cubature.cc

```diff
@@ -4407,83 +4417,93 @@ MolecularGrid::~MolecularGrid() {
         delete[] y_;
         delete[] z_;
         delete[] w_;
-        delete[] index_;
     }
 }
 
 void MolecularGrid::block(int max_points, int min_points, double max_radius) {
-    // Hack
-    Options &options_ = Process::environment.options;
-
-    // Reassign
     std::shared_ptr<GridBlocker> blocker;
-    if (options_.get_str("DFT_BLOCK_SCHEME") == "NAIVE") {
+    if (options_.blockscheme == "NAIVE") {
         blocker = std::shared_ptr<GridBlocker>(
-            new NaiveGridBlocker(npoints_, x_, y_, z_, w_, index_, max_points, min_points, max_radius, extents_));
-    } else if (options_.get_str("DFT_BLOCK_SCHEME") == "OCTREE") {
+            new NaiveGridBlocker(npoints_, x_, y_, z_, w_, max_points, min_points, max_radius, extents_));
+    } else if (options_.blockscheme == "OCTREE") {
         blocker = std::shared_ptr<GridBlocker>(
-            new OctreeGridBlocker(npoints_, x_, y_, z_, w_, index_, max_points, min_points, max_radius, extents_));
+            new OctreeGridBlocker(npoints_, x_, y_, z_, w_, max_points, min_points, max_radius, extents_));
+    } else if (options_.blockscheme == "ATOMIC") {
+        blocker = std::shared_ptr<GridBlocker>(new AtomicGridBlocker(npoints_, x_, y_, z_, w_, max_points, min_points,
+                                                                     max_radius, extents_, molecule_, atomic_grids_));
     }
 
-    blocker->set_print(options_.get_int("PRINT"));
-    blocker->set_debug(options_.get_int("DEBUG"));
-    blocker->set_bench(options_.get_int("BENCH"));
+    blocker->set_print(options_.print);
+    blocker->set_debug(options_.debug);
+    blocker->set_bench(options_.bench);
 
     blocker->block();
 
     delete[] x_;
     delete[] y_;
     delete[] z_;
     delete[] w_;
-    delete[] index_;
 
     x_ = blocker->x();
     y_ = blocker->y();
     z_ = blocker->z();
     w_ = blocker->w();
-    index_ = blocker->index();
 
     npoints_ = blocker->npoints();
     max_points_ = blocker->max_points();
     max_functions_ = blocker->max_functions();
     collocation_size_ = blocker->collocation_size();
-
+    size_t count = 0;
```

Comment:
Was `count` an artifact of a previous version of this code? It doesn't seem to be doing anything.

Reply:
Not sure anymore. Removed now.